modulemd/tests/test-modulemd-common.c

```diff
@@ -82,6 +82,84 @@ test_modulemd_load_file (void)
 }
 
 
+static void
+test_modulemd_load_string (void)
+{
+  const gchar *yaml_string = NULL;
+  g_autoptr (GError) error = NULL;
+  g_autoptr (ModulemdModuleIndex) idx = NULL;
+  g_autofree gchar *output = NULL;
+
+  /* This function is a wrapper around lower-level functions, so it should be
+   * okay to just test basic success and failure here.
+   */
+
+  /* Trivial modulemd */
+  g_clear_error (&error);
+  g_clear_object (&idx);
+  yaml_string =
+    "---\n"
+    "document: modulemd\n"
+    "version: 2\n"
+    "data:\n"
+    "  name: trivialname\n"
+    "  stream: trivialstream\n"
+    "  summary: Trivial Summary\n"
+    "  description: >-\n"
+    "    Trivial Description\n"
+    "  license:\n"
+    "    module: MIT\n"
+    "...\n";
+  idx = modulemd_load_string (yaml_string, &error);
+  g_assert_no_error (error);
+  g_assert_nonnull (idx);
+
+  /* Make sure loaded index dumps to string cleanly */
+  g_clear_error (&error);
+  g_clear_pointer (&output, g_free);
+  output = modulemd_module_index_dump_to_string (idx, &error);
+  g_assert_no_error (error);
+  g_assert_nonnull (output);
+
+
+  /* NULL string should raise an exception */
+  g_clear_error (&error);
+  g_clear_object (&idx);
+  modulemd_test_signal = 0;
+  signal (SIGTRAP, modulemd_test_signal_handler);
+  idx = modulemd_load_string (NULL, &error);
+  g_assert_cmpint (modulemd_test_signal, ==, SIGTRAP);
+  g_assert_null (idx);
+
+
+  // An empty string returns non-NULL but empty index. Is that right?
```

Comment:
That's a really good question, actually.

With the previous behavior (`ModuleIndex.new()` followed by `ModuleIndex.update_from_*()`), you would initialize an empty `ModuleIndex` first and then updating from an empty string would populate it with zero new entries. An empty string is valid YAML (it would appear as a YAML stream of zero documents). The same would be true of a zero-length file. So what you're getting here is the same as you would get there: an empty `ModuleIndex`. I think this is therefore correct.